File: crypto/src/main/java/io/oversec/one/crypto/images/xcoder/ImageXCoderFacade.java
package io.oversec.one.crypto.images.xcoder;

import android.content.Context;
import io.oversec.one.crypto.images.xcoder.blackandwhite.BlackAndWhiteImageXCoder;

import java.util.ArrayList;
import java.util.List;

public class ImageXCoderFacade {

    public static List<ImageXCoder> getAll(Context ctx) {
        ArrayList<ImageXCoder> res = new ArrayList<>();
        res.add(new BlackAndWhiteImageXCoder(ctx));
        return res;
    }
}

File: crypto/src/main/java/io/oversec/one/crypto/images/xcoder/ImageXCoderFacade.kt
package io.oversec.one.crypto.images.xcoder

import android.content.Context
import io.oversec.one.crypto.images.xcoder.blackandwhite.BlackAndWhiteImageXCoder
import java.util.ArrayList

object ImageXCoderFacade {

    fun getAll(ctx: Context): List<ImageXCoder> {
        return ArrayList<ImageXCoder>().also { it.add(BlackAndWhiteImageXCoder(ctx)) }
    }
}
